c7n/resources/vpc.py

```diff
@@ -2708,3 +2708,58 @@ class resource_type(query.TypeInfo):
         arn_type = 'traffic-mirror-target'
         universal_taggable = object()
         id_prefix = 'tmt-'
+
+
+@RouteTable.filter_registry.register('cross-az-nat-gateway-route')
+class CrossAZRouteTable(Filter):
+    """Filter route-table which is attached to a subnet that is not in the same availability zone (AZ) of the
+    NAT Gateway subnet AZ. This filter is applicable to those route-tables which have next hop as NatGateways for
+    any route. If a route-table has NAT gateway from us-west-2a and it is associated to subnet from us-west-2b then it
+    will flagged, and NatGatewayInCrossAvailabilityZone key would be set a True.
+
+    :Example:
+    .. code-block:: yaml
+            policies:
+              - name: cross-az-nat-gateway-traffic
+                resource: aws.route-table
+                filters:
+                    - type: cross-az-nat-gateway-route
+                actions:
+                  - notify
+    """
+    schema = type_schema('cross-az-nat-gateway-route')
+
+    def process(self, resources, event=None):
+        # dump of all subnets and nat-gateways to avoid multiple API calls
+        all_subnets = self.manager.get_resource_manager('aws.subnet').resources()
+        all_nat_gws = self.manager.get_resource_manager('nat-gateway').resources()
+        # Create subnet_id AZ mapping
+        subnets_az_map = {subnet["SubnetId"]: subnet["AvailabilityZone"] for subnet in all_subnets}
+        # Create nat_gateway subnet_id AZ mapping
+        nat_gws_subnet_map = {nat_gateway['NatGatewayId']: nat_gateway["SubnetId"] for nat_gateway in
+                              all_nat_gws}
+        # List of AZ where NAT gateways are created. To make sure NAT Gateway is available in a given AZ
+        all_nat_gw_az = [subnets_az_map[nat_gateway["SubnetId"]] for nat_gateway in
+                         all_nat_gws]
+        results = []  # To return filtered resources
+        for res in resources:
+            res['NatGatewayAvailabilityZone'] = {}
+            for route in res["Routes"]:
+                if route.get("NatGatewayId") and route.get("State") == "active":
```

Comment:
@vvishwakarma-intuit this code block is still pretty gnarly in terms of things being nested inside one another. One trick I usually use to avoid nested code is to avoid putting the main logic inside the if statement and instead, flip the logic on the if statement and then do `continue`. For example, I would convert this into
```
if not route.get("NatGatewayId") or route.get("State") != "active":
  continue

// Rest of code goes here without having to be inside the if statement
```

Reply:
Modified the code as suggested